### PR TITLE
Even more API v2

### PR DESCRIPTION
**Pull request by rowanseymour**
* Adds authenticate view for surveyor
* Adds endpoints for campaigns and campaign events

### CODE REVIEW

File: temba/api/models.py [hunk 1]
```diff
@@ -528,14 +537,14 @@ def get_orgs_for_role(cls, user, role):
         takes a single character role (A, E, S, etc) and maps it to a UserGroup.
         """
 
-        if role == 'A':
+        if role == cls.ROLE_ADMIN:
```
Comment (rowanseymour) [inline temba/api/models.py]: I guess I just love namespacing things...

File: temba/api/v2/urls.py [hunk 2]
```diff
@@ -2,15 +2,20 @@
...
 
 urlpatterns = [
     url(r'^$', api, name='api.v2'),
     url(r'^/explorer/$', ApiExplorerView.as_view(), name='api.v2.explorer'),
+    url(r'^/authenticate$', AuthenticateView.as_view(), name='api.v2.authenticate'),
+
+    # ========== endpoints A-Z ===========
     url(r'^/broadcasts$', BroadcastEndpoint.as_view(), name='api.v2.broadcasts'),
+    url(r'^/campaigns$', CampaignsEndpoint.as_view(), name='api.v2.campaigns'),
+    url(r'^/campaign_events$', CampaignEventsEndpoint.as_view(), name='api.v2.campaign_events'),
```
Comment (rowanseymour) [inline temba/api/v2/urls.py]: In v1 this is just `events.json` but now that we have `channel_events.json`, clearer to call this `campaign_events.json`

File: temba/api/v2/views.py [hunk 3]
```diff
@@ -82,6 +94,49 @@ def get_context_data(self, **kwargs):
...
+                    user.set_org(org)
+                    user.set_role(role)
+                    token = get_or_create_api_token(user)
+
+                    if token:
+                        tokens.append({'org': {'id': org.pk, 'name': org.name}, 'token': token})
+            else:
+                return HttpResponse(status=403)
+
+            return JsonResponse({'tokens': tokens})
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: v1 format here is just

```json
[
    {"id": 2733, "name": "Nyaruka RS", "token": "81133df0d4759e92b51ab3c35597d7d4d229dc3a"}
]
```

But have re-organised into an object for v2. Some would say returning an array from an API endpoint isn't best practice.

```json
{
    "tokens": [
        {"org": {"id": 2733, "name": "Nyaruka RS"}, "token": "81133df0d4759e92b51ab3c35597d7d4d229dc3a"}
    ]
}
```

Comment (ericnewcomer) [inline temba/api/v2/views.py]: Agreed. Certainly makes it easier to augment in the future.

**Comment by ericnewcomer** [comment]
yep

**Pull request merged**